network/p2p/gossip/bloom.go

```diff
@@ -8,35 +8,47 @@ import (
 
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/utils/bloom"
+	safemath "github.com/ava-labs/avalanchego/utils/math"
 )
 
-// NewBloomFilter returns a new instance of a bloom filter with at most
-// [maxExpectedElements] elements anticipated at any moment, and a false
-// positive probability of [falsePositiveProbability].
+// NewBloomFilter returns a new instance of a bloom filter with at least [minTargetElements] elements
+// anticipated at any moment, and a false positive probability of [targetFalsePositiveProbability]. If the
+// false positive probability exceeds [resetFalsePositiveProbability], the bloom filter will be reset.
 //
 // Invariant: The returned bloom filter is not safe to reset concurrently with
 // other operations. However, it is otherwise safe to access concurrently.
 func NewBloomFilter(
-	maxExpectedElements int,
-	falsePositiveProbability float64,
+	minTargetElements int,
+	targetFalsePositiveProbability,
+	resetFalsePositiveProbability float64,
 ) (*BloomFilter, error) {
-	bloom, err := bloom.New(bloom.OptimalParameters(
-		maxExpectedElements,
-		falsePositiveProbability,
-	))
+	numHashes, numEntries := bloom.OptimalParameters(
```

Comment:
We could attempt to bound `numEntries` here to something reasonable (like < 64KB)?

Reply:
For example, `bloom.OptimalParameters(100000, 0.01)` is `(7,119814)`: https://go.dev/play/p/QIzQ1dkOrsm

Reply:
We should probably leave it up to the caller to ensure this is limited to a reasonable number of entries.